geodesy/haversine_distance.py

```diff
@@ -0,0 +1,48 @@
+from math import asin, atan, cos, sin, sqrt, tan, pow, radians
+
+
+def haversine_distance(lat1: float, lon1: float, lat2: float, lon2: float) -> float:
+    """
+    Calculate great circle distance between two points in a sphere,
+    given longitudes and latitudes. 
+    (https://en.wikipedia.org/wiki/Haversine_formula)
+
+    Args:
+        lat1, lon1: latitude and longitude of coordinate 1
+        lat2, lon2: latitude and longitude of coordinate 2
+        
+    Returns: 
+        geographical distance between two points in metres
+
+    >>> int(haversine_distance(37.774856, -122.424227, 37.864742, -119.537521)) # From SF to Yosemite
+    254352
+
+    """
+
+    # CONSTANTS per WGS84 https://en.wikipedia.org/wiki/World_Geodetic_System
+    AXIS_A = 6378137.0
+    AXIS_B = 6356752.314245
+    RADIUS = 6378137
+
+    # Equation parameters
+    # Equation https://en.wikipedia.org/wiki/Haversine_formula#Formulation
+    flattening = (AXIS_A - AXIS_B) / AXIS_A
+    phi_1 = atan((1 - flattening) * tan(radians(lat1)))
+    phi_2 = atan((1 - flattening) * tan(radians(lat2)))
+    lambda_1 = radians(lon1)
+    lambda_2 = radians(lon2)
+
+    # Equation
+    sin_sq_phi = pow(sin((phi_2 - phi_1) / 2), 2)
+    sin_sq_lambda = pow(sin((lambda_2 - lambda_1) / 2), 2)
```

Comment:
Using `math.pow()` is very slow compared to pythons `**` notation for powers.

Reply:
What about sqrt vs ** .5?

Reply:
According to [this stackoverflow thread](https://stackoverflow.com/questions/327002/which-is-faster-in-python-x-5-or-math-sqrtx) you should use `math.sqrt`.

Reply:
And as a side note, we can simplify the above lines even more: `a * a` is a lot faster than `a ** 2`.

So you can change the above lines to `(sin(phi_2 - phi_1) / 2) * (sin(phi_2 - phi_1) / 2)` etc

Reply:
Did that and made it a little more readable